src/app/encoder.cpp

```diff
@@ -65,8 +65,7 @@ static uint16_t doEncodeApsFrame(BufBound & buf, uint16_t profileID, uint16_t cl
     }
     else
     {
-        result = buf.Fit() ? buf.Written() : 0;
-        CHECK_FRAME_LENGTH(result, "Buffer too small");
+        CHECK_FRAME_LENGTH((result = buf.Fit() ? buf.Written() : 0), "Buffer too small");
```

Comment:
Why this change?  Having a side-effect inside the condition there seems pretty hard to read to me...